quiz-microsevice/data-access/repositories/quiz-repository.js

```diff
@@ -61,4 +61,17 @@ const deleteQuiz = async (clientId, id) => {
   await Quiz.findOneAndDelete({ _id: id });
 };
 
-module.exports = { createQuiz, deleteQuiz };
+const retrieveQuizzes = async (clientId) => {
+
+  const { Quiz, Question} = getModelsForClient(clientId);
+  
+  const quizzes = await Quiz.find();
+  const questions = await Question.find(); 
```

Comment:
Retrieve only the quizzes don't retrieve it's questions, And use toDTO to map quizzes.